### DISCUSSION TITLE
ValueError() while using .predict() function

### DISCUSSION THREAD

**Discussion by sayedathar11** [awslabs/gluonts#1701] [Q&A]
I am using a Trained Predictor to make future predictions for 12 months , Earlier I have used make_evaluation_predictions() function on same testing_data and it worked properly , however now my aim is to make predictions on the unseen data that is future data and .predict() function is raising this error , can someone please help me where I am going wrong I am attaching error screenshot.
[](url
![gluonts_preds_error](https://user-images.githubusercontent.com/17905011/133962133-43080280-7f6e-4f6a-93b9-80f3e2bfb520.PNG)
)

**Comment by pratikgehlott** [comment]
Even, I am facing the same issue

**Reply by pratikgehlott** [reply]
@mbohlkeschneider. @lostella  can you guys help?

**Comment by lostella** [comment]
Looks like there are shape issues. Can you be more specific? What is the model, what does the data look like that you are providing to the `.predict` method (at least in terms of shapes). It’s not possible for anyone to help you unless you provide specifics of the issue that you are facing: what do you have (ie the mode) what are you giving it (the data) what do you expect vs what you are getting.

A tip for future issue reports of any kind: please **avoid screenshots** (unless for graphics like plots); copy-pasting code and error messages is just as quick and much more useful. Use triple-backticks to wrap code snippets (see [GitHub docs on proper text formatting](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax))

**Reply by pratikgehlott** [reply]
Hello @lostella, what I did is, I created the ListData for testing in the same as I did for training. But the ListData is working fine with make_evaluation_predictions

```
forecast_it, ts_it = make_evaluation_predictions(
    dataset=testing_data,  
    predictor=predictor,  
    num_samples= 2   
)

```


but when I do this with predictor.predict, I get the error.

`predictions = list(predictor.predict(testing_data))`



```
error =  ValueError: all the input array dimensions for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 110 and the array at index 2 has size 105 
```

**Comment by mbohlkeschneider** [comment]
Hi @pratikgehlott, does your dataset have dynamic features i.e. `feat_dynamic_real`? If this is the case, you need to provide them for the prediction range when using `DeepAREstimator`. `make_evaluation_predictions` will actually cut the last `prediction_length` steps of the target for evaluation purposes so this condition is satisfied. 

In other words: Your `feat_dynamic_real` need to be the length of the target + `prediction_length`.

**Reply by pratikgehlott** [reply]
okay, will try..thanks @mbohlkeschneider

**Reply by pratikgehlott** [reply]
Hi, @mbohlkeschneider I have one doubt, How are we suppose to know the value of `feat_dynamic_real` in the future, because as you are suggesting the `predictor.predict` needs the length of `feat_dynamic_real` to be the length of the target + `prediction_length`.

**Reply by pratikgehlott** [reply]
@mbohlkeschneider any comments?

**Reply by sayedathar11** [reply]
@lostella @mbohlkeschneider , I tried your suggestion and it worked , just needed why we should have length of feat_dynamic_real equals 
length of target + prediction_length , any reasons behind it why we do so ?

**Comment by mbohlkeschneider** [comment]
The dynamic time features need to be known for future values for many algorithms (such as DeepAR). The main rationale behind this is that these features are supposed to model known future events such as holidays or planned promotion actions. 

If you only have features for the past range, you could try `MQ-CNN` instead, which allows to train and forecast with only past dynamic features only.
